src/managers/CacheManager.js

```diff
@@ -0,0 +1,52 @@
+class CacheManager extends Map {
+    /**
+     * Creates a new CacheManager instance.
+     * @param {Number} ttl The time to live for the cache in minutes.
+     */
+    constructor(client, exports) {
+        super();
+
+        this.client = client;
+        this.exports = exports;
+
+        /** 
+         * The time to live for the cache in miliseconds.
+         * @private
+         */
+        this._ttl = this.client.options.cache.ttl * 60 * 1000;
+
+        this._reset();
+    }
+
+    /**
+	 * Searches for a single item where the given function returns a truthy value. This behaves like
+	 * [Array.find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find).
+     * @param {Function} f The function to test for each element.
+     * @returns {*} 
+     */
+    find(f) {
```

Comment:
Not a big fan of single-letter variables. Your call of course, just wanted to bring it up.